frontend/simple/views/components/Contribution/Contribution.vue

```diff
@@ -0,0 +1,171 @@
+        :aria-label="editAriaLabel"
+        @click="handleEditClick"
+      >
+        <i class="fa" :class="iconClass" aria-hidden="true"></i>
+      </button>
+    </div>
+  </li>
+
+  <li v-else-if="isUnfilled">
+    <button :class="itemClasses" @click="handleClick">
+      <slot></slot>
+    </button>
+  </li>
+
+  <li v-else :class="itemClasses">
+    <slot></slot>
+  </li>
+</template>
+<style lang="scss" scoped>
+@import "../../../assets/sass/theme/index";
+
+.c-contribution,
+.c-form {
+  min-height: 45px;
+}
+
+.c-contribution {
+  width: 100%;
+  font-size: inherit;
+
+  &.is-default {
+    border: none;
+    background-color: $white-ter;
+  }
+
+  &.is-editable {
+    border: none;
+    background-color: $primary-bg-a;
+  }
+
+  &.is-unfilled {
+    cursor: pointer;
+    text-align: left;
+
+    &:hover,
+    &:focus {
+      background-color: $primary-bg-a;
+    }
+  }
+
+  &-icon {
+    color: $primary;
+    margin-right: $gi-spacer-sm;
+  }
+}
+</style>
+<script>
+export default {
+  name: 'Contribution',
+  props: {
+    variant: {
+      type: String,
+      validator (value) {
+        return [
+          'default', // grey box just to read the content
+          'unfilled', // dashed box interactive to add a new contribution
+          'editable' // blue box to edit a contribution
+        ].indexOf(value) > -1
+      },
+      default: 'default'
+    }
+  },
+  data () {
+    return {
+      isAdding: false,
+      isEditing: false,
+      isFilled: null, // when true, show add/save button.
+      placeholders: [this.L('Portuguese classes'), this.L('Programming'), this.L('Cooking'), this.L('Parties'), this.L('Free cinema tickets')]
+    }
+  },
+  computed: {
+    itemClasses () {
+      return [
+        'box is-compact c-contribution',
+        `is-${this.variant}`,
+        { 'has-controls': this.isEditable }
+      ]
+    },
+    isEditable () {
+      return this.variant === 'editable'
+    },
+    isUnfilled () {
+      return this.variant === 'unfilled'
+    },
+    editAriaLabel () {
+      return this.L('Edit contribution settings')
+    },
+    randomPlaceholder () {
+      return this.placeholders[Math.floor(Math.random() * this.placeholders.length)]
+    },
+    iconClass () {
+      return this.$listeners.click ? 'fa-ellipsis-v' : 'fa-edit'
+    }
+  },
+  methods: {
+    handleClick () {
+      if (this.$listeners.click) {
+        this.$listeners.click()
+      } else {
+        this.isAdding = true
+      }
+    },
+    handleEditClick () {
+      if (this.$listeners.click) {
+        this.$listeners.click()
+      } else {
+        this.isEditing = true
+        this.isFilled = true
+      }
+    },
+    verifyValue (event) {
+      this.isFilled = !!event.target.value
+    },
+    cancel () {
+      this.isAdding = false
+      this.isEditing = false
+      this.isFilled = false
+    },
+    handleSubmit () {
+      this.$emit('new-value', this.$refs.input.value)
+      this.cancel()
```

Comment:
Needs to check if contents is empty or contains all whitespace. Use a regex or if you don't know how to do that put a TODO here to verify either manually or via vuelidate (ideally via vuelidate)

Reply:
It can be simpler than that using trim. Also I've added some `aria` attrs for a11y purposes :)